packages/hatchify-core/src/dataTypes/number/getFinalize.ts

```diff
@@ -33,6 +34,62 @@ export function getFinalize(
     control,
     orm: finalizeOrm(props.orm),
 
+    // Passed  - a possible "client" version of the data type.
+    //           - A value passed to createTodo()
+    //           - The serialized value returned from the server
+    // Returns - a valid "client" instance of the data type.
+    // Throws  - if the possible value is not valid.  ❓
+    // Example : "today" => new Date()
+    //         : throw "'4 $core' is not a valid date";
+    setClientPropertyValue: (userValue: UserValue) => {
+      return coerce(userValue, control)
+    },
+
+    // Passed  - a valid "client" version of the data type
+    // Returns - Something that can be serialized to JSON
+    // Example : new Date() => '2023-07-17T01:45:28.778Z'
+    serializeClientPropertyValue: (value: number | null): number | null => {
+      return value
+    },
+
+    // Passed  - Any crazy value the client might use to filter
+    // Returns - A value react-rest can understand / use
+    // Throws  - If the data is bad ❓
+    // Example : ?filter[age]=xyz ... xyz => throw "xyz is not a number";
+    setClientQueryFilterValue(queryValue: UserValue) {
+      return coerce(queryValue, control)
+    },
+
+    // Passed  - Any value someone might try to query by
+    // Returns - A STRING value that can be sent as part of the query string.
+    // Example : true => "true";
+    serializeClientQueryFilterValue(value: number | null): string {
+      return JSON.stringify(value)
+    },
+
+    // Passed  - Any crazy STRING value the client might send as GET
+    // Returns - A type the client can use
+    // Throws  - If the data is bad ❓. Also, what should the behavior be?
+    // Example : '2023-07-17T01:45:28.778Z' => new Date('2023-07-17T01:45:28.778Z')
+    setClientPropertyValueFromResponse(
+      jsonValue: ValueInRequest,
+    ): number | null {
+      if (jsonValue === null || jsonValue === undefined) {
+        if (control.allowNull !== false) {
+          return null
+        }
+        // todo: should we throw error if server is returning invalid values
+        throw new HatchifyCoerceError("as a non-null value")
+      }
+
+      if (isNaN(+jsonValue)) {
```

Comment:
cool, will move it right after the null/undefined checks inside of `coerce` and then I can remove the couple lines above here since they will be handled by `coerce` as well

Reply:
actually looks like this check is technically covered in coerce by 
```
  if (typeof value !== "number") {
    throw new HatchifyCoerceError("as a number")
  }
 ```
 the reason the `isNaN` check is in `setORMQueryFilterValue` is to convert string -> number before passing into coerce, will leave that one as is and just remove my `isNaN` check